### ISSUE TITLE
Swipe to delete history

### ISSUE THREAD

**Issue by person808** [mozilla-mobile/fenix#12610]
Broken out from #176 

### What is the user problem or growth opportunity you want to see solved?
Users should be able to swipe to delete history.

### How do you know that this problem exists today? Why is this important?
This is not possible currently. Deletion is done by tapping the three dot menu and selecting delete.

### Who will benefit from it?
Everyone who deletes items from history.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-3509)

**Comment by person808** [comment]
The problem with implementing this for the history page is that we are using version 2 of the paging library. We only have one view type because we cannot insert headers into the list of history items. This means we either 1) have to implement an `ItemTouchHelper` that only animates the history item view when swiped instead of the whole viewholder or 2) implement the gesture within the viewholder itself. Ideally we would upgrade to paging version 3 which supports inserting headers but it is still in alpha.

If either of the solutions above don't work we may have to wait for a stable version of paging v3.

**Comment by person808** [comment]
After multiple attempts, I think this may not be worth the effort to do right now (especially since my semester starts soon). The main issue I have run into is that if you undo deleting a history item, the item returns in a "swiped" state (the swipe background shows instead of the original item view). [Enabling](https://github.com/mozilla-mobile/fenix/blob/1823fdb66dec1be0f8b7e85b23f945ec48b171e3/app/src/main/java/org/mozilla/fenix/library/history/HistoryView.kt#L103) change animations fixes this, but because we currently rebind all visible viewholders, this introduces noticeable flickering.

A couple things I have tried:
1. Copying the default `ItemAnimator` and removing the alpha animation that causes the flickering. This introduces some other weirdness with the animations.
2. Being smarter about rebinding only the items that change. Fixes the swipe and undo issue, but the flickering is still present for those items and the code was rather messy.
3. Using `PagedListAdapter.submitList` to update the list since it uses `DiffUtil`. Turns out this method only runs when you pass it a new list (I think, my memory is a little fuzzy here) and honestly the flickering might still appear.

Things I have not tried:
1. Calling `notifyItemChanged` from the touch helpers `onSwiped` method, directly or indirectly. From implementing swipe to delete bookmarks it looks like this would trigger a "swipe back" animation which might fix the issue with swipe and undo. I'm unsure how this would interact with disabled change animations.
2. Using payloads so we can be smarter about rebinding history items. This might be able to fix the flickering.

If I had the time I would try both of these.
Regarding my previous comment: I discovered `ItemTouchCallback` has a `getDefaultUiUtil` method which makes it easy to only animate the actual history item view.
